src/App.tsx

```diff
@@ -1,26 +1,162 @@
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterTodo, setFilterTodo] = useState('All');
+  const [newTitleTodo, setNewTitleTodo] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState<number[]>([]);
+
+  const hideErrorMessage = () => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, []);
+
+  const addTodo = (newTodo: Omit<Todo, 'id'>) => {
+    setLoading(currentIds => [...currentIds, newTodo.userId]);
+    todoService
+      .postTodo(newTodo)
+      .then(addedTodo => {
+        setTodos(currentTodos => [...currentTodos, addedTodo]);
+        setNewTitleTodo('');
+        setLoading(currentIds =>
+          currentIds.filter(id => id !== newTodo.userId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setLoading(currentIds =>
+          currentIds.filter(id => id !== newTodo.userId),
+        );
+        hideErrorMessage();
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (userId: number) => {
+    setLoading(currentId => [...currentId, userId]);
+    todoService
+      .deleteTodo(userId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== userId),
+        );
+        setLoading(currentId => currentId.filter(id => id !== userId));
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        setLoading(currentId => currentId.filter(id => id !== userId));
+        hideErrorMessage();
+      });
+  };
+
+  const handleEmptyLineError = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = newTitleTodo.trim();
+
+    if (!trimmedTitle) {
+      setErrorMessage('Title should not be empty');
+      setTimeout(() => setErrorMessage(''), 3000);
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      title: trimmedTitle,
+      completed: false,
+      userId: todoService.USER_ID,
+    });
+
+    addTodo({
+      userId: todoService.USER_ID,
+      title: trimmedTitle,
+      completed: false,
+    });
+  };
+
+  const handleChangeNewTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitleTodo(event.target.value);
+  };
+
+  const handleDeleteAllCompleted = (todosId: number[]) => {
+    todosId.forEach(id => {
+      deleteTodo(id);
+    });
+  };
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      if (filterTodo === 'active') {
+        return !todo.completed;
+      }
+
+      if (filterTodo === 'completed') {
```

Comment:
Use a created enum here. Also, consider will be better if you use switch case here instead of if conditions
<img width="410" alt="image" src="https://github.com/user-attachments/assets/745ca271-d454-440f-b14a-cd0c8cfbd853" />